pkg/core/interop_system.go

```diff
@@ -182,7 +182,7 @@ func bcGetTransactionHeight(ic *interop.Context, v *vm.VM) error {
 // engineGetScriptContainer returns transaction that contains the script being
 // run.
 func engineGetScriptContainer(ic *interop.Context, v *vm.VM) error {
-	v.Estack().PushVal(stackitem.NewInterop(ic.Container))
+	v.Estack().PushVal(transactionToStackItem(ic.Container.(*transaction.Transaction)))
```

Comment:
I guess we could also have block here for `OnPersist`?

Reply:
Should we return an error in this case or return the block?

Reply:
C# node returns whatever it has, be it a block or transaction, so we should do the same probably. Although it would complicate things considerably.

Reply:
But at the same time we can keep `GetScriptContainer() blockchain.Transaction` signature for interop function, regular VM code will never get anything but transaction from it, it's only native contracts that can be run within block container (which I don't really like, honestly).

Reply:
I also modified test `engine.go` contract to check if all works correct, so it's OK now.